client/main.js

```diff
@@ -1,25 +1,85 @@
-import React from "react";
 import { Meteor } from "meteor/meteor";
+import React, { Component } from "react";
 import { render } from "react-dom";
-import { BaseForm } from "uniforms-antd";
-import { message } from "antd";
+import moment from "moment";
+import AutoForm from "uniforms-antd/AutoForm";
+import { ErrorsField, SubmitField } from "uniforms-antd";
+import DateRange from "./components/DatePicker";
+import TimeRange from "./components/TimePicker";
+import schema from "./lib/schema";
+import { ListField, ListItemField } from 'uniforms-antd';
+
+class App extends Component {
+  onChange = (...args) => {
+    console.log("model onchange", ...args);
+  };
+
+  transformModel = (mode, model) => {
+    // as antd returns and accepts moment object, transform the model before storing/validating in the model.
+
+    if (mode === "validate" || mode === "submit") {
+      const { dateRange, timeRange } = model;
+      let newModel = {};
+      newModel.dateRange = {};
+      newModel.timeRange = {};
+      // clone the object to prevent mutation
+      newModel.dateRange.start = dateRange.start;
+      newModel.dateRange.stop = dateRange.stop;
+      newModel.timeRange.start = timeRange.start;
+      newModel.timeRange.stop = timeRange.stop;
+
+
+      // convert moment to date object if the field value is not undefined.
+      newModel.dateRange.start = newModel.dateRange.start
+        ? newModel.dateRange.start.toDate()
+        : newModel.dateRange.start;
+      newModel.dateRange.stop = newModel.dateRange.stop
+        ? newModel.dateRange.stop.toDate()
+        : newModel.dateRange.stop;
+      newModel.timeRange.start = newModel.timeRange.start
+        ? newModel.timeRange.start.toDate()
+        : newModel.timeRange.start;
+      newModel.timeRange.stop = newModel.timeRange.stop
+        ? newModel.timeRange.stop.toDate()
+        : newModel.timeRange.stop;
+      return newModel;
+    }
+    // return the unaltered model in other scenarios.
+    return model;
+  };
 
-import TimePickerWrapper from "./TimePickerWrapper/timePickerWrapper";
-import DatePickerWrapper from "./DatePickerWrapper/datePickerWrapper";
-export default class CustomPicker extends React.Component {
   render() {
+    const { transformModel, onChange } = this;
     return (
-      <div>
-        <h2>Uniform Custom Field</h2>
-        <div className="custom-field-container">
-          <DatePickerWrapper />
-          <TimePickerWrapper />
-        </div>
+      <div className="container">
+        <h3>Uniform custom field form</h3>
+        <AutoForm
+          label={false}
+          schema={schema}
+          onChange={onChange}
+          spacing={3}
+          modelTransform={transformModel}
```

Comment:
The component should be make the transform inside itself

Reply:
Hi @cesarve77 , I have moved the transform into the component